tests/rados/test_node_drain_customer_bug.py

```diff
@@ -0,0 +1,206 @@
+"""
+The file contain the method to check the customer issue-
+ CEPH-83593996 - Check that the Ceph cluster logs are being generated appropriately according to the log level
+"""
+
+import random
+import re
+import time
+from threading import Thread
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.mgr_workflows import MgrWorkflows
+from ceph.rados.serviceability_workflows import ServiceabilityMethods
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83593996
+    Bug id - https://bugzilla.redhat.com/show_bug.cgi?id=2305677
+    1. Configure a cluster that have more than four OSD nodes
+    2. Select an OSD node and drain the node
+    3. Parallely execute the ceph mgr fail command
+    4. Check that exception occurs on the cluster
+    5. Perform the following workaround  steps-
+          5.1 ceph config-key rm mgr/cephadm/osd_remove_queue
+          5.2 ceph mgr fail
+    6. If exception not occured then check for the Traceback logs
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mgr_obj = MgrWorkflows(node=cephadm)
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    service_obj = ServiceabilityMethods(cluster=ceph_cluster, **config)
+    ceph_nodes = kw.get("ceph_nodes")
+
+    mgr_daemon = Thread(
+        target=background_mgr_task, kwargs={"mgr_object": mgr_obj}, daemon=True
+    )
+    osd_list = []
+
+    for node in ceph_nodes:
+        cmd_host_chk = f"ceph orch host ls --host_pattern {node.hostname}"
+        out = rados_obj.run_ceph_command(cmd=cmd_host_chk)
+        if not out:
+            log.info(f"The {node.hostname} is not in the cluster")
+            continue
+        if node.role == "osd":
+            node_osds = rados_obj.collect_osd_daemon_ids(node)
+            osd_list = osd_list + node_osds
+    osd_weight_chk = check_set_reweight(rados_obj, osd_list)
```

Comment:
With the removal failure in earlier test,it is expected that the weights be 0. 

Our test should be that we verify the removal after the failure + upgrade to fixed builds.

After the upgrade, we should try to remove the same host that we were trying to remove earlier, be it in any state.

This should not be a reason for test failure. This is what we are verifying here.

Reply:
I removed the code.